src/components/ContactPreview.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+
+import { css } from '../../styled-system/css'
+import useAvatar from '@/components/useAvatar'
+import { useRouter } from 'next/navigation'
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs'
+import { Database } from '@/types/supabase'
+import { useContext } from 'react'
+import { UserContext } from '@/components/UserProvider'
+import { Profile } from '@/types/models'
+
+type Props = {
+  contact: Profile
+}
+
+export default function ContactPreview({ contact }: Props) {
+  const supabase = createClientComponentClient<Database>()
+  const router = useRouter()
+  const { user } = useContext(UserContext)
+
+  const { AvatarPreview } = useAvatar({
+    avatarUrl: contact.avatar_url,
```

Comment:
bug: when I login, the page failed here, contact somehow was `undefined`.

Reply:
fixed